async_simple/coro/Collect.h

```diff
@@ -521,6 +520,35 @@ inline auto collectAny(LazyType<Ts>... awaitables) {
     return detail::CollectAnyVariadicImpl(std::move(awaitables)...);
 }
 
+// collectAny with std::pair<Lazy, CallbackFunction>
+template <typename... Ts>
+inline Lazy<size_t> collectAny(Ts... args) {
+    auto tuple = std::make_tuple(std::move(args)...);
+    auto lazy_pair = [&tuple]<size_t... I>(std::index_sequence<I...>) {
+        return std::make_pair(
+            collectAny(std::move(std::get<0>(std::get<I>(tuple)))...),
+            std::make_tuple(std::move(std::get<1>(std::get<I>(tuple)))...));
+    }(std::make_index_sequence<sizeof...(Ts)>());
+
+    auto result = co_await std::move(lazy_pair.first);
```

Comment:
This still introduces new coroutines, which may be a performance concern. Let's try to avoid the new creation of coroutines as much as possible.